zerver/lib/events.py

```diff
@@ -772,8 +772,13 @@ def _draft_update_action(i: int) -> None:
 
         if event["op"] == "add":
             person = copy.deepcopy(person)
-            if client_gravatar and person["avatar_url"].startswith("https://secure.gravatar.com/"):
-                person["avatar_url"] = None
+            if person["email_address_visibility"] != UserProfile.EMAIL_ADDRESS_VISIBILITY_EVERYONE:
+                client_gravatar = False
+            del person["email_address_visibility"]
```

Comment:
I suppose we have tests that verify `email_address_visibility` won't appear in the output event.

I'll also note that we could potentially have this code call a variant of `can_access_email_address`, since we do have the calling `user_profile` and thus their role.

Reply:
I think by "output event" you meant the user object in the state that is added/update using this event. Yes, the tests verify this. To be clear, the `email_address_visibility` is still present in the event sent to client as of now.

Reply:
Yeah, makes sense. We can resolve this item since whether `email_address_visibilty` should be in here is being discussed in the "api design" thread.